NPE when merge base is not found

When I set `compareToMergeBase` to true, and don't ensure that I have all the refs properly fetched, I'm getting an NPE. There should be an actionable error message instead.

```
org.apache.maven.MavenExecutionException: Failed to execute gitflow-incremental-builder.
    at com.vackosar.gitflowincrementalbuild.boundary.MavenLifecycleParticipant.perform (MavenLifecycleParticipant.java:94)
    at com.vackosar.gitflowincrementalbuild.boundary.MavenLifecycleParticipant.afterProjectsRead (MavenLifecycleParticipant.java:64)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:264)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:61)
Caused by: java.lang.NullPointerException
    at com.vackosar.gitflowincrementalbuild.control.DifferentFiles$Worker.getMergeBase (DifferentFiles.java:166)
    at com.vackosar.gitflowincrementalbuild.control.DifferentFiles$Worker.resolveReference (DifferentFiles.java:232)
    at com.vackosar.gitflowincrementalbuild.control.DifferentFiles$Worker.getBranchDiff (DifferentFiles.java:105)
    at com.vackosar.gitflowincrementalbuild.control.DifferentFiles$Worker.access$200 (DifferentFiles.java:87)
    at com.vackosar.gitflowincrementalbuild.control.DifferentFiles.get (DifferentFiles.java:64)
    at com.vackosar.gitflowincrementalbuild.control.ChangedProjects.get (ChangedProjects.java:42)
    at com.vackosar.gitflowincrementalbuild.boundary.UnchangedProjectsRemover.doAct (UnchangedProjectsRemover.java:107)
    at com.vackosar.gitflowincrementalbuild.boundary.UnchangedProjectsRemover.act (UnchangedProjectsRemover.java:70)
    at com.vackosar.gitflowincrementalbuild.boundary.MavenLifecycleParticipant.perform (MavenLifecycleParticipant.java:87)
    at com.vackosar.gitflowincrementalbuild.boundary.MavenLifecycleParticipant.afterProjectsRead (MavenLifecycleParticipant.java:64)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:264)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:61)
```

Looks like adding a check if commit is not null would be enough: https://github.com/gitflow-incremental-builder/gitflow-incremental-builder/blob/master/src/main/java/com/vackosar/gitflowincrementalbuild/control/DifferentFiles.java#L164

--- a/src/main/java/com/vackosar/gitflowincrementalbuild/control/DifferentFiles.java
+++ b/src/main/java/com/vackosar/gitflowincrementalbuild/control/DifferentFiles.java
@@ -162,7 +162,11 @@ public class DifferentFiles {
                 walk.markStart(walk.lookupCommit(baseCommit));
                 walk.markStart(walk.lookupCommit(referenceHeadCommit));
                 RevCommit commit = walk.next();
-                walk.close();
+                if (commit == null) {
+                    throw new IllegalStateException(String.format(
+                            "Cannot find merge base, try fetching more history.%n\tbase: %s%n\treference: %s",
+                            baseCommit, referenceHeadCommit));
+                }
                 logger.info("Using merge base of id: " + commit.getId());
                 return commit;
             }

